src/SonarScanner.MSBuild.Common/ProjectInfo/ProjectInfo.cs

```diff
@@ -103,30 +82,27 @@ public DirectoryInfo ProjectFileDirectory() =>
     public string ProjectGuidAsString() =>
         ProjectGuid.ToString("D", CultureInfo.InvariantCulture).ToUpperInvariant();
 
-    public string TryGetAnalysisFileLocation(AnalysisResultFileType fileType) =>
-        TryGetAnalyzerResult(fileType, out var result) ? result.Location : null;
-
     public FileInfo[] AllAnalysisFiles(ILogger logger)
     {
-        var compiledFilesPath = TryGetAnalysisFileLocation(AnalysisResultFileType.FilesToAnalyze);
-        if (compiledFilesPath is null
-            || !File.Exists(compiledFilesPath))
-        {
-            return [];
-        }
-
-        var result = new List<FileInfo>();
-        foreach (var path in File.ReadAllLines(compiledFilesPath))
+        if (FindAnalysisResultFile(AnalysisResultFileType.FilesToAnalyze)?.Location is { } filesToAnalyze && File.Exists(filesToAnalyze))
         {
-            try
+            var result = new List<FileInfo>();
+            foreach (var path in File.ReadAllLines(filesToAnalyze))
```

Comment:
Small possible optimization: `result` can be initialized with the right capacity with `File.ReadAllLines(filesToAnalyze).Length`. You could even allocate the result array upfront and `result[i]=new(path)` instead. Or use  `Select` and `ToArray()`.